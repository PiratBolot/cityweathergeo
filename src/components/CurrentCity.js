import React from "react";
import {getWeatherByCoords} from "../ApiHelper";

class CurrentCity extends React.Component {
    constructor(props) {
        super(props);
        this.state = { currentPosition: props.currentPosition };
    }

    async componentDidMount() {
        let data = await getWeatherByCoords(this.state.currentPosition.coords);
        let res = data.response;
        //let parsedData = parseData(res);
        this.setState({data: res, city: res.name + ", " + res.sys.country});
    }


    render = () => (
        <div className="CurrentCity">

            <div className="CityName">{this.state.city}</div>
        </div>
    )
}

export default CurrentCity;