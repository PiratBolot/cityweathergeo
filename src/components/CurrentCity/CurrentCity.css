.weather_frame {
    display: flex;
    flex-direction: row;
    color: #738287;
    background-color: white;
    margin: 5px;
    border-radius: 5px;
    box-shadow: 0 0 5px 0 #202020;
    flex-wrap: wrap;
    padding: 5px 0;
}

.weather_frame_now {
    flex-grow: 1;
    padding: 0px 7px;
}

.wn_body {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.city_name {
    font-size: 48px;
    font-weight: bold;
    text-align: start;
}

.weather_now {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    padding: 10px 15px 15px 0px;
}

.weather_icon {
    max-width: 150px;
    max-height: 150px;
    filter: drop-shadow(2px 4px 6px black);
    padding-right: 1px;
}

.temperature_c {
    font-size: 100px;
}

.wn_body {
    padding: 0 10px;
}

.wn_line {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 5px 10px;
    color: white;
    background-color: #a7afb1;
    margin: 5px 0;
    border-radius: 5px;
}

.info_label {
    font-weight: bold;
    margin-right: 20px;
}

.info_value {
    font-weight: bold;
    color: #435257;
}

.date {
    font-weight: bold;
    padding-top: 10px;
}

.weather_description {
    text-align: start;
    font-weight: bold;
    padding: 0px 0px;
}

.image_container_overlay {
    width: inherit;
    margin-left: auto;
    margin-right: auto;
    font-weight: bold;
}