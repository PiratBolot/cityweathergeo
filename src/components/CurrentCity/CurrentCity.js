import React from "react";
import {getWeatherByCoords, getWeatherByCityName, parseWeatherResponse} from "../../WeatherApi";
import './CurrentCity.css'
import CircularSpinner from "../PreLoader/CircularSpinner";

class CurrentCity extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            currentPosition: props.currentPosition,
            currentCity: props.currentCity,
            loaded: false
        };
    }

    async componentDidMount() {
        let data;
        if (this.state.currentCity) {
            data = await getWeatherByCityName(this.state.currentCity);
        } else {
            data = await getWeatherByCoords(this.state.currentPosition.coords);
        }
        let res = data.response;
        let parsedData = parseWeatherResponse(res);

        /* Uncomment if you want to see the preLoader explicitly
        var answer = prompt('question', 'defaultAnswer');
        */

        this.setState({loaded: !!res, data: res, parsedWeather: parsedData, city: res.name + ", " + res.sys.country});
    }


    render = () => (
        <div className="weather_frame">
            {
                this.state.loaded ?
                <div className="weather_frame_now">
                    <div className="city_name">{this.state.city}</div>
                    <div className="date">{new Date().toDateString()}</div>
                    <div className="weather_now">
                        <div className="temperature_c">
                            <span>{(this.state.data.main.temp - 273.14).toFixed(1) + "°C"}</span>
                        </div>
                        <img className="weather_icon" src={"http://openweathermap.org/img/wn/" +
                        this.state.data.weather[0].icon + "@2x.png"} alt=""
                        />
                        <div className="weather_description">{this.state.data.weather[0].description}</div>
                    </div>
                </div>
                    :
                    <div className="image_container_overlay">
                        <p className="label">Подождите, данные загружаются</p>
                        <CircularSpinner />
                    </div>
            }
            {
                this.state.parsedWeather ?
                    <div className="wn_body">
                        {
                            this.state.parsedWeather.map((e) =>
                                <div key={e.key} className="wn_line">
                                    <div className="info_label">{e.key}</div>
                                    <div className="info_value">{e.value}</div>
                                </div>
                            )
                        }
                    </div> : ""
            }
        </div>
    )
}

export default CurrentCity;