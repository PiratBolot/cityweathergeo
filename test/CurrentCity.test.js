import React from 'react';
import {mount} from '../enzyme/enzyme';
import {applyMiddleware, createStore} from "redux";
import {Provider} from "react-redux";
import rootReducer from "../src/reducers/Reducers";
import thunk from "redux-thunk";
import toJson from "enzyme-to-json";
import CurrentCity from "../src/components/currentCity/CurrentCity";

describe('Current City', () => {
    const initialState = {
        trackedCities: ['Syktyvkar', 'Moscow', 'Kiev'],
        errorMessage: {isError: false, errorMsg: ""},
        geolocation: {}
    };
    let store,wrapper;
    beforeEach(()=>{
        store = createStore(rootReducer, initialState, applyMiddleware(thunk));
        wrapper = mount( <Provider store={store}><CurrentCity /></Provider> );
    })
    it('+++capturing Snapshot of CurrentCity', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});