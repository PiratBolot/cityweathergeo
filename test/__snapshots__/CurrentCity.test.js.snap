// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current City +++capturing Snapshot of CurrentCity 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(CurrentCity)>
    <CurrentCity
      geolocation={Object {}}
      getLocation={[Function]}
    >
      <div
        className="current_city"
      >
        <Loader>
          <div
            className="image_container_overlay"
          >
            <p
              className="label"
            >
              Подождите, данные загружаются
            </p>
            <CircularSpinner>
              <div
                className="loader"
              >
                <svg
                  className="circular"
                  viewBox="25 25 50 50"
                >
                  <circle
                    className="path"
                    cx="50"
                    cy="50"
                    fill="none"
                    r="20"
                    strokeMiterlimit="10"
                    strokeWidth="2"
                  />
                </svg>
              </div>
            </CircularSpinner>
          </div>
        </Loader>
      </div>
    </CurrentCity>
  </Connect(CurrentCity)>
</Provider>
`;
